app/src/userpages/modules/permission/reducer.js

```diff
@@ -31,7 +43,14 @@ export default function (state: PermissionState = initialState, action: Permissi
                     ...state.byTypeAndId,
                     [(action.resourceType: string)]: {
                         ...(state.byTypeAndId[action.resourceType] || {}),
-                        [action.resourceId]: action.permissions.map(({ operation }) => operation),
+                        [action.resourceId]: (
+                            newPermissions(
+                                state,
+                                action.resourceType,
+                                action.resourceId,
+                                action.permissions.map(({ operation }) => operation).sort(),
```

Comment:
Is the sort needed?

Reply:
`newPermissions` compares arrays of strings using `Array#join`.